src/SEnergyCorrelator.ana.h

```diff
@@ -125,7 +138,45 @@ namespace SColdQcdCorrelatorAnalysis {
 
   }  // end 'DoLocalCalcWithPackage(double)'
 
-
+  void SEnergyCorrelator::DoLocalCalcManual(uint64_t jetIdx, vector<fastjet::PseudoJet> cstPseudoJet){
+    //get Jet info for manual calculations
+    const double jet_pT = m_input.jets[jetIdx].GetPT();
+    const double jet_Eta = m_input.jets[jetIdx].GetEta();
+    const double jet_Phi = m_input.jets[jetIdx].GetPhi();
+    const double jet_Ene = m_input.jets[jetIdx].GetEne();
+    ROOT::Math::PtEtaPhiEVector VecJet(jet_pT, jet_Eta, jet_Phi, jet_Ene);
+    TVector3 pJet(VecJet.X(), VecJet.Y(), VecJet.Z());
```

Comment:
Then following the above suggestion, this calculation could be moved up into the jet loop.

Reply:
Naturally, this means that you won't be able to fill both the transverse and total energy histograms in the same calculation. Instead, the user will need some way to tell the code what to use as weights. Personally, I prefer this approach as it makes it much more explicit about what calculations are going to be run.

One way you might implement this is to create a function which takes in a `PseudoJet` object (and maybe has an optional argument at the end which provides the axis the energy should be transverse or otherwise to), and calculates the corresponding quantity (total energy, pT, eT).

This function could then be called both in the jet loop where you 1st calculate the normalization for the weights, and then on each of the constituents in the manual calculation.